### ISSUE TITLE
Didn't get too far

### ISSUE THREAD

**Issue by craigbruenderman** [ciscops/ansible-viptela#1]
<img width="1402" alt="Screen Shot 32" src="https://user-images.githubusercontent.com/1125996/56628432-bb0d4200-6617-11e9-97c6-d0eafdb381f7.png">
I poked through, setup vmanage variable and credentials, and tried to run a simple 

- include_tasks: facts.yml

TASK [ansible-viptela : Get Tenant Status] ********************************************************************************************************************
fatal: [10.0.0.2 -> localhost]: FAILED! => {"allow": "POST, OPTIONS", "cache_control": "no-cache, no-store, must-revalidate", "changed": false, "connection": "close", "content": "", "content_length": "0", "date": "Wed, 24 Apr 2019 02:29:36 GMT", "msg": "Status code was 405 and not [200]: HTTP Error 405: Method Not Allowed", "redirected": false, "status": 405, "strict_transport_security": "max-age=31536000; includeSubDomains", "url": "https://10.0.0.2/dataservice/system/device", "vary": "Accept-Encoding", "x_frame_options": "DENY"}

Any idea what I'm missing?

**Comment by stevenca** [comment]
That facts tasks was left in there by mistake.  The only tasks that are tested are in the README.  You can this role in use here: https://github.com/ciscops/viptela-workshop/blob/master/README_AUTOMATION.md
I am starting to convert _some_ of the tasks to modules this week, but leaving as a role.  I'd be interested in knowing what your use case is to see if its there or in the plan.

**Comment by craigbruenderman** [comment]
Fair enough, I'll ignore it. My immediate use case is to make spinning up virtualized test environments easier - it's very painful to do from scratch, particularly all the CA stuff. As an SP, we need to be able to spin up and down environments quickly and control our whitelists directly, but there's not a good mechanism for that.

I'd also like to develop playbooks for stamping out our standard feature and device templates. I've seen some of the SE's use a Postman collection to populate a ton of standard templates quickly, but I've been unable to get ahold of it so far.

My target could be vSphere or AWS. I'm happy to contribute.

**Comment by stevenca** [comment]
That is exactly what that viptela-workshop does (i.e. spin up test environments), it just does it in VIRL.  It does not currently do templates/policy, but I am working on that now.  The role does not care _where_ the control plane instances are deployed, so if they are on VMware it should work.  I just do not have the tooling to build it on vmware (although it is on my long todo list).  This work will manifest itself at Cisco Live in session DEVNET-2098 (https://www.ciscolive.com/us/learn/sessions/session-catalog.html?search=DEVNET-2098#/) and session DEVWKS-2179 (https://www.ciscolive.com/us/learn/sessions/session-catalog.html?search=DEVWKS-2179#/).  Even if you do not plan to attend, the material will be accessible and you can will be able to access the lab in developer.cisco.com (it will be there by CLUS).  Have you taken a look through the workshop, does it look like it covers your requirements (assuming templates policy will be added)?

**Comment by craigbruenderman** [comment]
I think I'll have to adapt it for vSphere and AWS at least for things like inventory generation, which seems to be VIRL specific right now. I'm registered for 2179, and I'll try to get in 2098, which is currently full. I'm still working my way through the code, but so far it looks useful.

**Comment by stevenca** [comment]
The inventory is not VIRL specific.  It works on AWS as well: https://github.com/ciscops/viptela-ops.  In the case of this repo, I do not use the AWS dynamic inventory script because I create the tenants as CFT templates with tags.  I can then just get the IPs from the CFT templates and add them to the inventory dynamically.  This approach works well on "cloudy" things (e.g. AWS, Azure, OpenStack/VIRL, NFVIS), but not as well on something like VMware.  It would not be too hard to adapt, however, you just need to make sure that the items in the inventory get into the appropriate groups and their inventory have the appropriate key/value pair.  Basically, you just get the management information from the VMware dynamic inventory script and pull the rest from the yml files. 

Unfortunately, you'll have trouble using the viptela-ops repo with AWS, though, because it requires access to the AMIs for vmanage and vsmart.  I know it works because we use it internally for a few projects, but I could not push everything out into the open, so it is not complete.  The intent was to make it a generic framework for spinning viptela control planes up on things, I just haven't gotten back to it.  I'll probably finish it off after CLUS for another project.  In any case, it shows an example of how to do it.
